opentelemetry-operator.yaml

```diff
@@ -0,0 +1,130 @@
+            -X github.com/open-telemetry/opentelemetry-operator/internal/version.autoInstrumentationPython=$(grep -v '\#' versions.txt | grep autoinstrumentation-python | awk -F= '{print $2}')
+            -X github.com/open-telemetry/opentelemetry-operator/internal/version.autoInstrumentationDotNet=$(grep -v '\#' versions.txt | grep autoinstrumentation-dotnet | awk -F= '{print $2}')
+            -X github.com/open-telemetry/opentelemetry-operator/internal/version.autoInstrumentationGo=$(grep -v '\#' versions.txt | grep autoinstrumentation-go | awk -F= '{print $2}')
+            -X github.com/open-telemetry/opentelemetry-operator/internal/version.autoInstrumentationApacheHttpd=$(grep -v '\#' versions.txt | grep autoinstrumentation-apache-httpd | awk -F= '{print $2}')
+            -X github.com/open-telemetry/opentelemetry-operator/internal/version.autoInstrumentationNginx=$(grep -v '\#' versions.txt | grep autoinstrumentation-nginx | awk -F= '{print $2}')
+    test:
+      pipeline:
+        - runs: |
+            main --help || true # test tends to fail
+
+  - name: ${{package.name}}-otel-allocator-compat
+    description: "Compatibility package to place binaries in the location expected by upstream helm charts"
+    pipeline:
+      - runs: |
+          # The helm chart expects the binaries to be in / instead of /usr/bin
+          mkdir -p "${{targets.contextdir}}"
+          ln -sf /usr/bin/main ${{targets.contextdir}}/main
+
+update:
+  enabled: true
+  github:
+    identifier: open-telemetry/opentelemetry-operator
+    strip-prefix: v
+
+test:
+  environment:
+    contents:
+      packages:
+        - kubectl
+        - git
+        - mkcert
+        - kustomize
+        - curl
+  pipeline:
+    - uses: test/kwok/cluster
+    - name: Fetch the testdata from the source repo
+      runs: git clone --depth=1 https://github.com/open-telemetry/opentelemetry-operator/
+    - name: Install CRDs
+      working-directory: opentelemetry-operator/config/crd/
+      runs: |
+        kubectl wait --for=condition=Ready nodes --all
+        kustomize build | kubectl apply --server-side=true -f -
+        # wait for all CRDs to be created
+        sleep 3
+    - name: "Test manager run"
+      uses: test/daemon-check-output
+      with:
+        setup: |
+          mkdir -p /tmp/k8s-webhook-server/serving-certs/
+          mkcert -install
+          mkcert -key-file /tmp/k8s-webhook-server/serving-certs/tls.key -cert-file /tmp/k8s-webhook-server/serving-certs/tls.crt localhost
+        start: /usr/bin/manager  --metrics-addr=127.0.0.1:8080
+        timeout: 30
+        expected_output: |
+          Starting Controller
+        post: |
+          sleep 3
+          STATUS=$(curl -k -s -o /dev/null -w "%{http_code}" http://localhost:8081/healthz)
+          # Check if the status code is 200, if so exit 0, otherwise exit 1
+          if [ "$STATUS" -eq 200 ]; then
```

Comment:
You could use `curl --fail` and if would exits 1 if the response doesn't have 2xx code